### DISCUSSION TITLE
"changelog entry is empty for crate:" warning with a `CHANGELOG.md` present

### DISCUSSION THREAD

**Discussion by mkatychev** [GitoxideLabs/cargo-smart-release#23] [Q&A] [answered]
Hello, I am attempting to use `cargo smart-release --dry-run-cargo-publish --update-crates-index rex-sm` and I keep getting the warning below for [this commit](https://github.com/knox-networks/rex-sm/tree/5b783b9e0d737ea216d8d781f6992d4ad32d1062):
```
[INFO ] Updating crates-io index
[INFO ] WOULD auto-bump provided package 'rex-sm' from 0.5.0 to 0.5.1 for publishing
[WARN ] rex-sm: The following tags were not encountered during commit graph traversal: refs/tags/v0.4.1
[INFO ] WOULD modify existing changelog for 'rex-sm'.
[INFO ] Up to 1 changelog would be previewed if the --execute is set and --no-changelog-preview is unset.
[WARN ] WOULD stop release after commit as the changelog entry is empty for crate: rex-sm
[WARN ] To fix the changelog manually, run: cargo changelog --write rex-sm
```

I have a `CHANGELOG.md` present and created with `cargo changelog --write` and it is not clear where the warning is coming from.

**Comment by Byron** [comment, marked as answer]
This means that the changelog is fully generated and does not have any explicit content, like the one it would add when it detected commits with `fix|chore|feat|…:` subjects.

To fix it, write the changelog and enter some information about the release manually, and commit. Typically that's where I write `A maintenance release without user-facing changes.` directly under the `### Unreleased` section headline.

**Reply by mkatychev** [reply]
That makes sense, perhaps stating it as "No content for ## Unreleased" would clarify things

**Reply by Byron** [reply]
A PR for improving the wording would definitely be welcome.
